### ISSUE TITLE
Bug: <Reloaded claims VC redist isn't installed> when <it is on the Steam Deck>

### ISSUE THREAD

**Issue by MrCamel999** [Reloaded-Project/Reloaded-II#506]
**Describe the Bug**

**To Reproduce**
Steps to reproduce the behaviour:
1. Install VC_redist.x64.exe and the 86 exe
2. Open Reloaded
3. Reloaded complains about things not being installed

**Expected Behaviour**
The program shouldn't complain about things not being installed, because they are already installed.

**Screenshots**
I have a screen recording github won't let me upload. I'll try commenting on this with the video

**Additional Context**

**Comment by Sewer56** [comment]
The installer `Setup-Linux.exe` should have auto installed these.

If you are instead preferring to install manually, the old school way, make sure you are installing the redist into the same WINEPREFIX that Reloaded is installed to, i.e. same game in protontricks.

**Comment by MrCamel999** [comment]
https://github.com/user-attachments/assets/a8d76e4b-2b6a-4397-ab6b-42906b63a6ed
Figured out how to upload the video. GitHub simply hates mkv files for some reason

**Comment by Sewer56** [comment]
Oh, strange. That's a new one, I've not seen that happen before.

In any case, if the loader itself works as expected, you can ignore the message.

**Comment by MrCamel999** [comment]
Unfortunately, the loader doesn't work as expected. Trying to enable a mod causes a crash. I have two other installations on the system for P3R and P5R where this issue doesn't exist, and I updated those today to the same version, so I have zero clue what's different lol

**Comment by MrCamel999** [comment]
I can try using this fancy new Linux installer though, I encountered this while setting up Metaphor mods for the first time, so I'm not actually losing anything by just trying that. I'll report back on if it works magically

**Comment by MrCamel999** [comment]
![Image](https://github.com/user-attachments/assets/f0c11feb-248a-47b9-bcd5-92c1e11dc249)
I may just not understand how to use the linux installer, but it seemingly just is broken lol

**Comment by MrCamel999** [comment]
Thanks for the tip. Will try doing that and report back. I'll also make sure to record my entire process of installing everything to the prefix and stuff that way I can show *exactly* what I'm doing

**Comment by MrCamel999** [comment]
https://github.com/user-attachments/assets/84303e82-a73e-48fb-9a00-41b16e3b4ddd
New video. New prefix after deleting that folder in compatdata. New release download just to make sure. Same issue.

**Comment by MrCamel999** [comment]
I uploaded the wrong video hang on I'm dumb

**Comment by Sewer56** [comment]
Oh actually, one more thing you can try.
You have the 2019 redist, try the 2022 one:

https://aka.ms/vs/17/release/vc_redist.x64.exe
https://aka.ms/vs/17/release/vc_redist.x86.exe

(I'm on my phone but I believe that should be the 2022 one)

Maybe the automated GitHub build built it on a newer toolset and I didn't notice. Or I didn't notice a toolset bump on recent PR that bumped dependencies.

**Comment by MrCamel999** [comment]
Thanks. Those are in fact the 2022 ones. The 2019 ones actually come from those download buttons that appear in Reloaded II. Might be good to eventually update that. Unfortunately, that didn't fix the issue. Still working on getting that video uploaded. Was transferring the video to my laptop via discord but this video was too big for discord so now I have to sign into things on the deck and it's a pain typing these passwords bc I'm too lazy to sign into the password manager atm lol. Should be uploaded soon though.

**Comment by MrCamel999** [comment]
Failed to upload on github. I uploaded it to proton drive to transfer it to my laptop anyways, so I'll just link that. https://drive.proton.me/urls/6W8CVKZ7M0#oeVi0YEk589x

If you want me to upload it to yt instead of downloading random file from internet stranger just let me know and ill do it

**Comment by MrCamel999** [comment]
https://youtu.be/VJ7R3Z7AU18

Uploaded to YouTube anyways, not too hard to do that

**Comment by Sewer56** [comment]
You're fine. It's just weird it doesn't detect it, unless Metaphor is doing something really special with its WINEPREFIX setup.

Do still check if Reloaded boots with ASI Loader, if it's a false positive, you can ignore the message.

---

Also irrelevant but you shouldn't need .NET 7 or 8 SDKs, just 9.0.0 is enough.

**Comment by MrCamel999** [comment]
I'll be honest, I completely forget how to use ASI Loader lol. I feel like it's impacting something since I can't enable Metaphor mods in R2 with this error. Without the error in P3R it allows me to enable/disable just fine, so while I could be wrong, I'm pretty sure it's causing an issue.

I think I know where to find an ASI Loader tutorial to reference so I'll check.

Thanks for telling me I can delete versions 7 and 8, that makes this easier for when I eventually delete this prefix and try again lol

**Comment by MrCamel999** [comment]
I also plan on not using the ASI Loader and creating a seperate thing in steam to launch the game such that I have a modded and not modded way to watch just in steam. Do you know if you can undo deploying the asi loader into the game?

**Comment by Sewer56** [comment]
The approach in the guide you're using under 'Steam Configuration' could/would work too. In that case ASI Loader wouldn't be needed.

**Comment by MrCamel999** [comment]
Can confirm launching the game works through that Steam Configuration method. That's good at least. Haven't had any mods enabled but I assume that's a good thing regardless

**Comment by Sewer56** [comment]
Yup. Stuff works, just missing runtime dialog shows as false positive, weird.

**Comment by MrCamel999** [comment]
I wonder why I can't enable mods then. That's then weird as well

**Comment by Sewer56** [comment]
If you get a console window (provided you haven't disabled it), things work as expected.

Mods which are filled in are automatically enabled as needed, these don't need manual enable/disable.

**Comment by MrCamel999** [comment]
https://github.com/user-attachments/assets/3391717b-b9f7-4258-beae-d286b9588dca
This is what I mean. This happens when I try to enable any mod

**Comment by Sewer56** [comment]
It auto canceled a config save? What? That's very weird, all of this is kinda strange.

Maybe switch Proton version, not even I know fully what's going on anymore.

**Comment by MrCamel999** [comment]
I'll try switching proton versions then yeah. I think I have experimental running on P3R, so that is in fact a potential difference! Will report on how that goes

**Comment by MrCamel999** [comment]
Something has been screwed up bad. I can't get Reloaded to launch at all with that prefix now. I guess it's time to delete the prefix and try again lol

**Comment by MrCamel999** [comment]
Redoing the prefix on a new proton version (experimental) allows me to enable mods.

Thanks for all the help!!

A bit annoying about that false positive, but eh, I'll get over it. The main thing is that it works!

**Comment by dreamsyntax** [comment]
@MrCamel999 Just to confirm, did using 2022 redists make the warning go away?

**Comment by MrCamel999** [comment]
I never got the Linux Setup exe working, so that could be a solution for me if I were to get it working. It works though now, and it seems I'm the only one with this issue, so it's probably fine

**Comment by Sewer56** [comment]
I borked the setup exe yesterday by accident actually.
That said, the error they got with it was unrelated to the borking I did. They had a weird connection error.

**Comment by MrCamel999** [comment]
😂 I have made such a cursed issue lol

**Comment by Sewer56** [comment]
I'll close this as I have since reworked setup a tiny bit, both in launcher and out launcher.
And original user had issue fixed.
